### PR TITLE
Insertion sort

### PR DESCRIPTION
**Pull request by Ailyinchick**
I've completed base implementation of Insertion sort algorithm. Also checkout structure of project folders. Please leave a comment

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,10 @@
 bin/
+config
+description
 .classpath
+head
+hooks/
+info/
 .project
 <<<<<<< HEAD
```
Comment (dreva-dev) [inline .gitignore:9]: Resolve conflicts

File: src/Entities/ComparableEntity.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package Entities;
+
+import Interfaces.MyComparable;
+
+public abstract class ComparableEntity<T> implements MyComparable<T> {
```
Comment (dreva-dev) [inline src/Entities/ComparableEntity.java:5]: Why you need this abstract class?

File: src/Entities/Entity.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+	
+//  Methods
+	@Override
+	public boolean isGreater(Entity t1) {
+		return this.getAge() > t1.getAge();
```
Comment (dreva-dev) [inline src/Entities/Entity.java:38]: Add null check

File: src/Entities/Sorter.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package Entities;
+
+import Interfaces.InsertionSort;
+import Interfaces.SelectionSort;
+
+public class Sorter<T extends ComparableEntity<T>> implements InsertionSort<T>, SelectionSort<T> {
+
+	@Override
+	public boolean greaterThan(T t1, T t2) {
+		return t1.isGreater(t2);
```
Comment (dreva-dev) [inline src/Entities/Sorter.java:10]: Add null check

File: src/Interfaces/InsertionSort.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package Interfaces;
+
+import Entities.ComparableEntity;
+
+public interface InsertionSort<T extends ComparableEntity<T>> extends Sortable<T> {
+	
+	// Ascending insertion sort
+	public default void insertionSortAsc(T[] arr) {
```
Comment (dreva-dev) [inline src/Interfaces/InsertionSort.java:8]: Add null check here and in similar places

File: src/Interfaces/SelectionSort.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package Interfaces;
+
+import Entities.ComparableEntity;
+
+public interface SelectionSort<T extends ComparableEntity<T>> extends Sortable<T> {
+
+	    // Ascending selection sort
+		public default void selectionSortAsc(T[] arr) {
```
Comment (dreva-dev) [inline src/Interfaces/SelectionSort.java:8]: Add null check here and in similar places

**Comment by dreva-dev** [comment]
Please, use spaces instead of tabs

**Comment by dreva-dev** [comment]
use lower case for package names

**Comment by Ailyinchick** [comment]
New refactored branch with fixes and Maven created.

**Pull request closed without merging**